src/hardware/pcspeaker.cpp

```diff
@@ -420,109 +430,108 @@ void PCSPEAKER_SetType(bool pit_clock_gate_enabled, bool pit_output_enabled)
 	}
 }
 
-static unsigned output_buffer_length = 0;
-static double *output_buffer = NULL;
-static double *sampled_impulse = NULL;
-
 // TODO: check if this is accurate
-static inline double sinc(double t)
+static inline float sinc(float t)
```

Comment:
This is an interesting one. Do we have some specific reason for preferring 32-bit floats to 64-bit doubles?

I haven't personally done measurements lately, but I've seen quite a few devs claiming that doubles are a _lot_ faster on Intel, almost twice, at the very least for audio plugins, where the memory bandwidth requirements are negligible, but usually there's some heavy calculation going on. The situation might be quite different with video processing which is memory bandwidth heavy as well.

This sums it up pretty well:
https://stackoverflow.com/a/4584664

In the mixer stuff I'm just using floats because SpeeX doesn't support doubles. But I've been wondering if it would be better to just use doubles everywhere. Perhaps it wouldn't really matter either way for what we're doing...

Reply:
It does make a difference in video and audio; even if the actual instruction takes the same number of cycles, the compiler can pack the calculations in SIMD calls (and get twice as many done), and when reading values from arrays, they only take up half the space so the effectiveness of L1 / L2 cache goes twice as far. 
Also, with 32 bits, that's a lot of extra resolution when handling values that we'll ultimately cast back down to 16 bit. Even if we're doing lots of sequencial math on them, the accumulated errors still won't cause a single rounding error. 
(@kklobe -- I remember you even added comparison to confirm this, when you were working on the mixer 👍 ). 
The performance is measureable on slower systems, and helps especially in things like FluidSynth.